pkg/gofr/datasource/kv-store/badger/logger.go

```diff
@@ -0,0 +1,42 @@
+package badger
+
+import (
+	"fmt"
+	"io"
+	"regexp"
+	"strings"
+)
+
+type Logger interface {
+	Debug(args ...interface{})
+	Debugf(pattern string, args ...interface{})
+	Log(args ...interface{})
+	Logf(pattern string, args ...interface{})
+	Error(args ...interface{})
+	Errorf(patter string, args ...interface{})
+}
+
+type QueryLog struct {
+	Query    string `json:"query"`
+	Duration int64  `json:"duration"`
+	Keyspace string `json:"keyspace,omitempty"`
+}
+
+func (ql *QueryLog) PrettyPrint(writer io.Writer) {
+	fmt.Fprintf(writer, "\u001B[38;5;8m%-32s \u001B[38;5;206m%-6s\u001B[0m %8d\u001B[38;5;8mµs\u001B[0m %s\n",
+		clean(ql.Query), "CASS", ql.Duration, clean(ql.Keyspace))
+}
+
+// clean takes a string query as input and performs two operations to clean it up:
+// 1. It replaces multiple consecutive whitespace characters with a single space.
+// 2. It trims leading and trailing whitespace from the string.
+// The cleaned-up query string is then returned.
+func clean(query string) string {
+	// Replace multiple consecutive whitespace characters with a single space
+	query = regexp.MustCompile(`\s+`).ReplaceAllString(query, " ")
```

Comment:
Regexp should be moved outside the function in a var, this way it's computed only once and not each time the function is called

Reply:
I am refraining from defining a variable globally, need to think of something else

Reply:
OK

Reply:
These regexp.MustCompile() should be outside the function, this is a pattern in most external datasources and needs to be fixed in a separate PR. I will be creating an issue